release.js

```diff
@@ -0,0 +1,520 @@
+#!/usr/bin/env node
+const config = require('./releaseConfig')
+const yn = require('yn')
+const ora = require('ora')
+const readline = require('readline')
+const fs = require('fs')
+const { exec, spawn } = require('child_process')
+const util = require('util')
+const promiseExec = util.promisify(exec)
+const chalk = require('chalk')
+
+const { VERSION } = process.env
+
+let safeToClearWorkspace = false
+let rcNumber = NaN
+let versionRC = null
+let isFirstReleaseIteration = false
+
+//////////////////////////////
+// Release Helper functions //
+/////////////////////////////
+
+const stepTitle = message => {
+  console.log()
+  console.log(chalk.magenta('~'.repeat(message.length + 4)))
+  console.log(chalk.magenta(`| ${message} |`))
+  console.log(chalk.magenta('~'.repeat(message.length + 4)))
+  console.log()
+}
+
+const question = query => {
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout
+  });
+
+  console.log()
+  return new Promise(resolve => rl.question(`\n${query} (y/n) `, answer => {
+    const answerAsBoolean = yn(answer) || false
+    resolve(answerAsBoolean)
+    rl.close()
+  }))
+}
+
+const getNumberInput = query => {
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout
+  });
+
+  return new Promise(resolve => rl.question(`\n${query}`, answer => {
+    const answerAsNumber = parseInt(answer, 10)
+    rl.close()
+    resolve(answerAsNumber)
+  }))
+  .then((answerAsNumber) => {
+    if (!answerAsNumber) {
+      console.log(`❌ That was not a valid integer. Please type a valid integer 🙄 \n`)
+      return getNumberInput(query)
+    }
+    return answerAsNumber
+  })
+}
+
+const proceedYesNo = async query => {
```

Comment:
I think it would be easier if these util functions would be moved to a separate file. That way it's easier to read the mean of the release logic, without any UI util functions in it.

Reply:
I agree. I did that in one of the previews commit but I had some problem sharing variables across files. I think I will just write a temporary file and delete it after when we exit the script